tests/locale/keymap_or_locale.pm

```diff
@@ -0,0 +1,97 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Keyboard layout test in console and gdm after boot
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use utils;
+
+sub verify_default_keymap_textmode {
+    my ($test_string, $tag, %tty) = @_;
+    select_console($tty{console}) if defined($tty{console});
+    type_string($test_string);
+    assert_screen($tag);
+    # clear line in order to add user bernhard to tty group
+    # clear line to avoid possible failures in following console tests if scheduled
+    send_key("ctrl-w");
+}
+
+sub list_locale_settings {
+    my $program = shift;
+
+    x11_start_program($program) if (get_var('DESKTOP') && (!check_var('DESKTOP', 'textmode')));
+    script_run('localectl status');
```

Comment:
your goal here to give info about current state of localectl to a person who will review the test results, right ?
And your achieve it, no doubts. But your approach has two disadvantages :
1. Result of command execution will be displayed in screenshot, so no ability to copy them as a text
2. x11_start_program is really expensive and error prone way of doing something which you need to use only if you really want to test some behavior related to it. 
I suggest to do the following : 
```
select_console('root-console');
assert_script_run('localectl status > /tmp/localectl.status.out');
upload_logs('/tmp/localectl.status.out');

```

Reply:
good suggestion. Even better use `$self->save_and_upload_log('localectl status', '/tmp/localectl.status.out');`